core/storage/exploration/gae_models.py

```diff
@@ -1113,3 +1137,124 @@ def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
+    state and the exploration metadata at a particular version of the
+    exploration. The information about each state includes the version
+    number of the exploration on which the state was previously edited,
+    the name of the state at the previous version and the id of the user who
+    committed those changes. For metadata, the information includes the
+    version number of the exploration on which the metadata was previously
+    edited and the id of the user who committed those changes.
+
+    A new instance of this model is created each time a new exploration
+    is created or some changes are saved in an exploration.
+
+    The id of the model is generated as follows:
+    {exploration_id}-{exploration_version}
+    """
+
+    # The id of the corresponding exploration.
+    exploration_id = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The version of the corresponding exploration.
+    exploration_version = datastore_services.IntegerProperty(
+        required=True, indexed=True)
+    # The details of the previous commit on each state at a particular
+    # version of the exploration. The json structure will look like the
+    # following:
+    # {
+    #   [state_name: str]: {
+    #     "previously_edited_in_version": int,
+    #     "state_name_in_previous_version": str,
+    #     "committer_id": str
+    #   }
+    # }
+    # The json object can have multiple keys in this case depending on
+    # the number of states.
+    state_version_history = datastore_services.JsonProperty(
+        default={}, indexed=False)
+    # The exploration version on which the metadata was previously edited.
+    # If its value is v, then it will indicate that the metadata was modified
+    # when the exploration was updated from version v -> v + 1.
+    # During the creation of a new exploration, its value will be 0.
+    metadata_last_edited_version_number = datastore_services.IntegerProperty(
+        required=True, indexed=True)
+    # The user id of the user who committed the latest changes on the
+    # exploration metadata.
+    metadata_last_edited_committer_id = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The user ids of the users who did the 'previous commit' on each state
+    # in this version of the exploration. It is required during the
+    # wipeout process to query for the models efficiently.
+    committer_ids = datastore_services.StringProperty(
+        indexed=True, repeated=True)
+
+    @classmethod
+    def get_instance_id(cls, exp_id: str, exp_version: int) -> str:
+        """Returns ID of the exploration version history model.
+
+        Args:
+            exp_id: str. The ID of the exploration.
+            exp_version: int. The version of the exploration.
+
+        Returns:
+            str. A string containing exploration ID and
+            exploration version.
+        """
+        return 'version-history-%s.%s' % (exp_id, exp_version)
```

Comment:
Why do we have the prefix (`version-history-`)?

Reply:
It is just to distinguish a version history model id from other ids. For example, the id of SnapshotMetadataModel is of the form (id-version).

This is also done in case of ExplorationCommitLogEntryModel whose id is of the form (exploration-exp_id-exp_version)

Reply:
But this is a different model, we have many models that have the same ID as long as they are of different type. For `ExplorationCommitLogEntryModel` we sometimes prefix with `'exploration'` and sometimes with `'rights'` depending on what are we exactly doing so that is a bit different case.

Reply:
This prefix does not play any actual role in the functionality of the model. We can or cannot keep this prefix. Do you want me to remove it?

Reply:
I would remove it I don't see any benefit of it.

Reply:
Done.